### ISSUE TITLE
Throwing exception when using Content-Security-Policy "script-src 'self' " header

### ISSUE THREAD

**Issue by Kasta11111** [arivera12/BlazorDownloadFile#26]
Version: 2.1.6
AspNetCore: 3.1.8

After adding `Content-Security-Policy "script-src 'self' "` header DownloadFile function started throwing the following exception:

```
Could not find '_blazorDowloadFileBase64String' in 'window'.
Error: Could not find '_blazorDowloadFileBase64String' in 'window'.
    at http://192.168.0.30:5000/_framework/blazor.server.js:8:30944
    at Array.forEach (<anonymous>)
    at p (http://192.168.0.30:5000/_framework/blazor.server.js:8:30904)
    at http://192.168.0.30:5000/_framework/blazor.server.js:8:31614
    at new Promise (<anonymous>)
    at e.beginInvokeJSFromDotNet (http://192.168.0.30:5000/_framework/blazor.server.js:8:31587)
    at http://192.168.0.30:5000/_framework/blazor.server.js:1:20052
    at Array.forEach (<anonymous>)
    at e.invokeClientMethod (http://192.168.0.30:5000/_framework/blazor.server.js:1:20022)
    at e.processIncomingData (http://192.168.0.30:5000/_framework/blazor.server.js:1:18006)
```

**Comment by arivera12** [comment]
As far I can see you are using blazor server side right?

Are you doing cross site scripting?

Can you share any code sample which reproduces this error?

**Comment by Kasta11111** [comment]
Yes, I am using blazor server.

Yes, it's because of XSS.

The example project is attached below.
Headers are added in `Startup.cs` on line `49` with `app.UseMiddleware<SecurityHeadersMiddleware>();`.

[BlazorDownloadTest.zip](https://github.com/arivera12/BlazorDownloadFile/files/5868381/BlazorDownloadTest.zip)

**Comment by arivera12** [comment]
Ok I will take a look soon.

**Comment by arivera12** [comment]
@Kasta11111 I found what is wrong. Since blazor dowload file uses eval, setting Content-Security-Policy "script-src 'self' " disables the eval function. This would need some rewrite.

**Comment by Kasta11111** [comment]
Any progress?

**Comment by arivera12** [comment]
Not yet, in a few weeks

**Comment by arivera12** [comment]
https://github.com/arivera12/BlazorDownloadFile/commit/0df3baad527878e5fbc69919207b62b3d1347225

Now you should include the blazor download file script in this way 

`<script src="_content/BlazorDownloadFile/BlazorDownloadFileScript.js"><script>`

Available on version 2.3.1
